docker/api/exec_api.py

```diff
@@ -65,4 +64,15 @@ def exec_start(self, exec_id, detach=False, tty=False, stream=False):
         res = self._post_json(
             self._url('/exec/{0}/start', exec_id), data=data, stream=stream
         )
+        return res
+
+    @utils.minimum_version('1.15')
+    def exec_start(self, exec_id, detach=False, tty=False, stream=False):
+        res = self._exec_start(exec_id, detach=detach, tty=tty, stream=stream)
         return self._get_result_tty(stream, res, tty)
+
+    @utils.minimum_version('1.15')
+    def exec_start_socket(self, exec_id, detach=False, tty=False,
+                          stream=False):
+        res = self._exec_start(exec_id, detach=detach, tty=tty, stream=stream)
+        return self._get_raw_response_socket(res)
```

Comment:
Makes sense to me. To be honest, I actually vote for `socket=False` since there is already an option for asking for stream. On top of it, you'll get this feature for free in future (for new similar API calls).